src/commands/data/schemas/util/schemaverify.ts

```diff
@@ -135,61 +134,82 @@ export class SchemaVerify {
-            errors.push(`Property '${key}' is defined in the creation transition properties but not in the schema properties`);
-            ok = false;
-            return;
-          }
+    if (errors.length) {
+      ok = false;
+    }
 
-          // Check if the condition property is incorrectly typed vs the schema properties
-          if (!isEqual(schemaProperties[key], conditionProperties[key])) {
-            errors.push(`Property '${key}' has different type definitions in the creation transition and schema properties`);
-            ok = false;
-          }
-        });
-      });
+    return { ok, errors };
+  }
 
-      for (const [index, condition] of this.schema.creationTransition.conditions.entries()) {
-        if (condition.type !== 'input') {
-          continue;
-        }
+  validateTransition(transition: any, name: string) {
+    const errors = [];
+    const conditions = transition.conditions || [];
+
+    for (const [index, condition] of Object.entries<any>(conditions)) {
+      if (condition.type === 'input') {
         condition.configuration.$schema = 'http://json-schema.org/draft-07/schema#';
-        if (!this.ajv.validateSchema(condition.configuration)) {
-          ok = false;
+
+        const isValidConfiguration = this.ajv.validateSchema(condition.configuration);
+        if (!isValidConfiguration) {
           errors.push(...transformAjvErrors(`/creationTransition/conditions[${index}]/configuration`, this.ajv.errors));
         }
+
+        const result = this.validateAgainst(condition.configuration.properties, this.schema.properties, '');
+        result.forEach((error: string) => errors.push(`Transition - ${name} : property ${error}`));
       }
     }
 
-    /* check all conditions of the other transitions */
-    if (this.schema.transitions?.length) {
-      for (const [tIndex, transition] of this.schema.transitions.entries()) {
-        if (!transition.conditions) {
-          continue;
+    return errors;
+  }
+
+  validateAgainst(source: any, target: any, path: string) {
+    const invalidPaths = new Set([]);
+
+    // If the provided source object is not an object the property tree has been exhausted, thus return
+    if (typeof source !== 'object') {
+      return invalidPaths;
+    }
+
+    for (const key of Object.keys(source)) {
+      const sourceProperty = source[key];
+      const targetProperty = target?.[key];
+
+      /**
+       * If the source key is type and the value is a string we have reached a type definition
+       * If the property is not a string most likely the user has defined a property with the name of type
+       */
+      if (key === 'type' && typeof sourceProperty === 'string') {
+        // If the type property exists at the given path in the target, check if the types match
+        if (targetProperty && sourceProperty !== targetProperty) {
+          invalidPaths.add(`'${path}.type' is defined in both conditions and properties but is of the incorrect type`);
```

Comment:
(minor, suggestion) This message felt a bit weird, would something like `'${path}.type' in the condition does not match the same property in the schema` make sense?

Reply:
How bou dis? `'${path}.type' does not match the value found in the schema properties`